scraper.py

```diff
@@ -121,6 +130,32 @@ def get_posts(candidates):
             posts[candidate['post_id']]['known_candidates'] += 1
     return posts
 
+
+def get_sopn_date(result):
+    slug = result['election_id']
+    territory = result['organisation']['territory_code']
+
+    country = {
+        "ENG": Country.ENGLAND,
+        "WLS": Country.WALES,
+        "SCT": Country.SCOTLAND,
+        "NIR": Country.NORTHERN_IRELAND,
+    }
+
+    if slug.startswith("local") and territory not in country:
+        return None
+
+    if slug.startswith("local") and territory is not None:
+
+        (_, date_of_poll) = type_and_poll_date(slug)
+
+        return SOPN_PUBLISH_DATE.local(
+            date_of_poll, country=country[territory]
+        )
+    else:
+        return SOPN_PUBLISH_DATE.for_id(slug)
```

Comment:
Given this call to `for_id()` can raise an `AmbiguousElectionIdError`, can we handle that case.

I reckon if we don't know (and its hard to work out, for whatever reason), its preferable to `return None` here rather than completely fall over: if we don't know the SOPN date for some subset of ballots for some reason, that's no worse than what we have now.

Reply:
That's a good shout.

The logic here is pretty much a copy-paste of what's in WCIVF - that's probably a signal that this logic can be pulled out somewhere like the library.